provider/server/server_suite_test.go

```diff
@@ -123,7 +124,7 @@ var _ = BeforeSuite(func() {
 		return isSocketAvailable(opts.Address)
 	}).WithTimeout(30 * time.Second).WithPolling(500 * time.Millisecond).Should(Succeed())
 
-	address, err := machine.GetAddressWithTimeout(3*time.Second, fmt.Sprintf("unix://%s", opts.Address))
+	address, err := machine.GetAddressWithTimeout(3*time.Second, filepath.Join("unix://", opts.Address))
```

Comment:
The use of filepath.Join for constructing file paths on the filesystem is common practice. However, since `unix://` is a URL scheme rather than a file path component, using filepath.Join for it is not suitable.

Reply:
Agree! reverted